File: onyx-database/src/com/onyx/descriptor/IdentifierDescriptor.java
package com.onyx.descriptor;

import com.onyx.persistence.annotations.IdentifierGenerator;

/**
 * Created by timothy.osborn on 12/11/14.
 */
public class IdentifierDescriptor extends IndexDescriptor
{
    protected IdentifierGenerator generator;

    public IdentifierGenerator getGenerator()
    {
        return generator;
    }

    public void setGenerator(IdentifierGenerator generator)
    {
        this.generator = generator;
    }

}

File: onyx-database/src/main/kotlin/com/onyx/descriptor/IdentifierDescriptor.kt
package com.onyx.descriptor

import com.onyx.persistence.annotations.values.IdentifierGenerator

/**
 * Created by timothy.osborn on 12/11/14.
 *
 * General information about a key identifier for an entity
 */
data class IdentifierDescriptor(
        var generator: IdentifierGenerator = IdentifierGenerator.NONE,
        override var name: String = "",
        override var type: Class<*> = Any::class.java,
        override var loadFactor: Byte = 1
) : IndexDescriptor(),BaseDescriptor {

    override lateinit var entityDescriptor: EntityDescriptor

    override fun hashCode(): Int = super.hashCode() + generator.hashCode()
    override fun equals(other: Any?): Boolean = super.equals(other) && (other as IdentifierDescriptor).generator == generator
}
